sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/handler/WebSocketsConnectionHandler.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.implementation.handler;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.azure.proton.transport.ws.impl.WebSocketImpl;
+import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.impl.TransportInternal;
+
+/**
+ * Creates an AMQP connection using web sockets (port 443).
+ */
+public class WebSocketsConnectionHandler extends ConnectionHandler {
+    static final int HTTPS_PORT = 443;
+
+    // This is the current limitation of https://github.com/Azure/qpid-proton-j-extensions
+    // once, this library enables larger frames - this property can be removed.
+    static final int MAX_FRAME_SIZE =  4 * 1024;
+
+    private static final String SOCKET_PATH = "/$servicebus/websocket";
```

Comment:
I didn't expect that path, but it checks out against the .NET track one code.

![](https://media3.giphy.com/media/K6VhXtbgCXqQU/giphy.gif)

Reply:
Me neither. It's a magical path.